### PR TITLE
added check for image storage

### LINKED ISSUE #111: Compose fails due to image storage being full
When building images with osbuild eventually the image storage will be full. We need a way to determine if it's at capacity before we create a new image. This way we won't have to wait for compose to error out due to storage issues.

Another idea is to have a way to delete / empty images on build. 

`clear_images: true`  // remove all images prior to building
or 
`clear_images: 3` // the oldest 3 images are to be removed prior to building

### PR DESCRIPTION
**Pull request by resoluteCoder**
Fixes #111 

- Adds check if image storage is full based on threshold percentage
- Adds ability to remove all images when var is true

### CODE REVIEW

File: roles/builder/tasks/main.yml [hunk 1]
```diff
@@ -1,4 +1,27 @@
...
+      register: output
+      changed_when: false
+
+    - name: Remove each image by UUID
+      ansible.builtin.command: "/usr/bin/composer-cli compose delete {{ (item | split)[0] }}"
+      loop: "{{ output.stdout_lines }}"
+      changed_when: true
+
+- name: Get current image storage size
+  ansible.builtin.shell: "set -eo pipefail df -h /var/lib/osbuild-composer/artifacts | tail -n 1 | awk '{ print $5; }' | rev | cut -c2- | rev"
```
Comment (maxamillion) [inline roles/builder/tasks/main.yml:16]: this concerns me but I can't think of a better way to accomplish this ..... at a minimum we should add a [`failed_when`](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html#defining-failure) clause on it or else we could have this variable not contain an integer and the type cast below will fail in unpleasant ways.

File: roles/builder/defaults/main.yml [hunk 2]
```diff
@@ -18,3 +18,5 @@ builder_compose_customizations:
     groups:
       - "users"
       - "wheel"
+builder_image_storage_threshold: 3 # percentage
```
Comment (maxamillion) [inline roles/builder/defaults/main.yml:21]: It's not clear to me what this variable's intended purpose is. Please add some information to the README. :)

File: roles/builder/tasks/main.yml (hunk 1 above)
Comment (resoluteCoder) [reply roles/builder/tasks/main.yml:16]: @maxamillion I agree lol I couldn't think of a better way. I was thinking about doing it in a python module but surprisingly it was easier in ansible. 

I'll add `failed_when` and write the docs

**Review by maxamillion — CHANGES_REQUESTED** [review]
This change needs a [changelog snippet](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_changelogs.html) now that we're doing those from now on. :)

**Review by matoval — APPROVED** [review]

**Pull request merged**